carl/envs/carl_env_new.py

```diff
@@ -0,0 +1,253 @@
+from __future__ import annotations
+
+import inspect
+from typing import Any, SupportsFloat
+
+from gymnasium import Wrapper, spaces
+import gymnasium
+from gymnasium.core import Env
+
+import jax
+from jax import numpy as jp
+import numpy as np
+import abc
+
+from carl.utils.types import Contexts, Context
+from carl.context.selection import AbstractSelector, RoundRobinSelector
+from carl.context.context_space import (
+    ContextFeature,
+    ContextSpace,
+    UniformFloatContextFeature,
+    ContextSampler,
+    NormalFloatContextFeature,
+)
+
+
+class CARLEnv(Wrapper, abc.ABC):
+    def __init__(
+        self,
+        env: Env,
+        contexts: Contexts | None = None,
+        state_context_features: list[str]
+        | None = None,  # list the context features which should be added to the state # TODO rename to obs_context_features?
+        obs_context_as_dict: bool = True,  # TODO discuss default
+        context_selector: AbstractSelector | type[AbstractSelector] | None = None,
+        context_selector_kwargs: dict = None,
+        **kwargs,
+    ):
+        super().__init__(env)
+
+        self.state_observation_space: gymnasium.spaces.Space = env.observation_space
```

Comment:
This could be named better. Base obs space? standard obs space?

Reply:
Agreed
- [ ] rename